paper-api/src/main/java/io/papermc/paper/world/damagesource/FallLocation.java

```diff
@@ -0,0 +1,65 @@
+package io.papermc.paper.world.damagesource;
+
+import net.kyori.adventure.translation.Translatable;
+import org.jetbrains.annotations.ApiStatus;
+import org.jspecify.annotations.NullMarked;
+
+/**
+ * Represents a type of location from which the entity fell
+ *
+ * @since 1.21.4
+ */
+@NullMarked
+@ApiStatus.Experimental
+public sealed interface FallLocation extends Translatable permits FallLocationImpl {
```

Comment:
FallLocationType maybe? Rn it sounds like this is actually holding a location

Reply:
Yes, the Type makes more sense